File: 1689. partitioning-into-minimum-number-of-deci-binary-numbers.java
class Solution {
    public int minPartitions(String n) {
        int ans = 0;
        for (int i = 0; i < n.length(); ++i) {
            ans = Math.max(ans, n.charAt(i) - '0');
        }
        return ans;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1:
        String n1 = "32";
        int output1 = sol.minPartitions(n1);
        System.out.println("Example 1: Input: n = " + n1);
        System.out.println("Output: " + output1);
        System.out.println("Expected: 3");
        System.out.println();
        
        // Example 2:
        String n2 = "82734";
        int output2 = sol.minPartitions(n2);
        System.out.println("Example 2: Input: n = " + n2);
        System.out.println("Output: " + output2);
        System.out.println("Expected: 8");
        System.out.println();
        
        // Example 3:
        String n3 = "27346209830709182346";
        int output3 = sol.minPartitions(n3);
        System.out.println("Example 3: Input: n = " + n3);
        System.out.println("Output: " + output3);
        System.out.println("Expected: 9");
    }
}

File: 1689. partitioning-into-minimum-number-of-deci-binary-numbers.py
class Solution:
    def minPartitions(self, n: str) -> int:
        return int(max(n))

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1:
    n = "32"
    output = sol.minPartitions(n)
    print(f"Example 1: Input: n = {n}")
    print(f"Output: {output}")
    print(f"Expected: 3")
    print()
    
    # Example 2:
    n = "82734"
    output = sol.minPartitions(n)
    print(f"Example 2: Input: n = {n}")
    print(f"Output: {output}")
    print(f"Expected: 8")
    print()
    
    # Example 3:
    n = "27346209830709182346"
    output = sol.minPartitions(n)
    print(f"Example 3: Input: n = {n}")
    print(f"Output: {output}")
    print(f"Expected: 9")